File: spark2kudu/src/main/scala/Spark2Kudu.scala
import java.util.Date

import scala.collection.JavaConverters._
import scala.collection.immutable.IndexedSeq
import com.google.common.collect.ImmutableList
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.kudu.ColumnSchema.ColumnSchemaBuilder
import org.apache.kudu.client.KuduClient.KuduClientBuilder
import org.apache.kudu.client.{CreateTableOptions, KuduClient, KuduTable}
import org.apache.kudu.spark.kudu.KuduContext
import org.apache.kudu.{Schema, Type}
import org.apache.kudu.spark.kudu._
import org.apache.kudu.client._
import collection.JavaConverters._

object Spark2Kudu {
  case class Customer(name:String, age:Int, city:String)

  def main(args: Array[String]): Unit = {


    var sc: SparkContext = null
    var kuduClient: KuduClient = null
    var table: KuduTable = null
    var kuduContext: KuduContext = null

    val tableName = "spark-test"

    val appID = new Date().toString + math.floor(math.random * 10E4).toLong.toString

    val conf = new SparkConf().
      setMaster("local[*]").
      setAppName("test").
      set("spark.app.id", appID)
    kuduClient = new KuduClientBuilder("172.172.241.228:7051").build()
    sc = new SparkContext(conf)
    kuduContext = new KuduContext("172.172.241.228:7051",sc)

    val sqlContext = new org.apache.spark.sql.SQLContext(sc)
    val df = sqlContext.read.options(Map("kudu.master" -> "172.172.241.228:7051", "kudu.table" -> "app_event_aggre_mem_user_kudu_da"))
      .kudu.limit(2000)

    df.show(100)
     println("****************")
    df.schema.foreach(p=>println(p))


    if(kuduContext.tableExists("spark_test_table"))
      {
        kuduContext.deleteTable("spark_test_table")
      }
    kuduContext.createTable(
      "spark_test_table", df.schema, Seq("fuid"),
      new CreateTableOptions()
        .setNumReplicas(3)
        .addHashPartitions(List("fuid").asJava, 3))

    kuduContext.insertRows(df, "spark_test_table")

  }


}

File: spark2kudu/src/main/java/com/haiziwang/spark/Spark2Kudu.java
package com.haiziwang.spark;
import org.apache.kudu.ColumnSchema;
import org.apache.kudu.Schema;
import org.apache.kudu.Type;
import org.apache.kudu.client.CreateTableOptions;
import org.apache.kudu.client.KuduClient;
import org.apache.kudu.client.KuduException;
import org.apache.kudu.client.KuduTable;
import org.apache.kudu.spark.kudu.KuduContext;
import org.apache.spark.SparkConf;
import org.apache.spark.SparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

import java.util.*;

import scala.collection.Seq;


public class Spark2Kudu {
    public static void main(String[] args) {
        SparkContext sc  = null;

        KuduTable table = null;
        KuduContext kuduContext  = null;
        String master = "172.172.241.228:7051";
        String tableName = "spark-test-java";
        String appID = "Spark2Kudu"+System.currentTimeMillis();
        SparkConf conf = new SparkConf().
                setMaster("local[*]").
                setAppName("test").
                set("spark.app.id", appID);
        sc = new SparkContext(conf);
        kuduContext = new KuduContext(master,sc);
        SQLContext sqlContext = new org.apache.spark.sql.SQLContext(sc);
        Map confmap= new HashMap<String, String>();
        confmap.put("kudu.master", master);
        confmap.put("kudu.table", "app_event_aggre_mem_user_kudu_da");
        KuduClient kuduClient = new KuduClient.KuduClientBuilder("172.172.241.228:7051").build();

        DataFrame df = sqlContext.read().format("org.apache.kudu.spark.kudu").options(confmap).load().limit(1000)
                .select("fuid","femail");
        df.show(100);
        System.out.println(df.schema());
        if(kuduContext.tableExists(tableName))
        {
            kuduContext.deleteTable(tableName);
        }

        ArrayList list = new ArrayList<String>();
        list.add("fuid");
        List<ColumnSchema> columns = new ArrayList(2);
        columns.add(new ColumnSchema.ColumnSchemaBuilder("fuid", Type.INT64)
                .key(true)
                .build());
        columns.add(new ColumnSchema.ColumnSchemaBuilder("femail", Type.STRING)
                .nullable(true)
                .build());
        Schema schema = new Schema(columns);
        try {
            kuduClient.createTable(tableName, schema,
                    new CreateTableOptions().setRangePartitionColumns(list));

        } catch (KuduException e) {
            System.out.println("create kudu failed");
            e.printStackTrace();
        }

        kuduContext.insertRows(df, tableName) ;


    }
}
